src/home/homework-luks.c

```diff
@@ -1826,6 +1826,67 @@ static int luks_format(
         return 0;
 }
 
+static int make_partition(
+                const char *label,
+                sd_id128_t uuid,
+                struct fdisk_partition *p) {
+        _cleanup_(fdisk_unref_parttypep) struct fdisk_parttype *t = NULL;
+        int r;
+
+        t = fdisk_new_parttype();
+        if (!t)
+                return log_oom();
+
+        r = fdisk_parttype_set_typestr(t, SD_GPT_USER_HOME_STR);
+        if (r < 0)
+                return log_error_errno(r, "Failed to initialize partition type: %m");
+
+        r = fdisk_partition_set_type(p, t);
+        if (r < 0)
+                return log_error_errno(r, "Failed to set partition type: %m");
+
+        r = fdisk_partition_set_name(p, label);
+        if (r < 0)
+                return log_error_errno(r, "Failed to set partition name: %m");
+
+        r = fdisk_partition_set_uuid(p, SD_ID128_TO_UUID_STRING(uuid));
+        if (r < 0)
+                return log_error_errno(r, "Failed to set partition UUID: %m");
+
+        return 0;
+}
+
+static int query_partition_offset_and_size(
+                struct fdisk_context *c,
+                int partno,
+                uint64_t *ret_offset,
+                uint64_t *ret_size) {
+        _cleanup_(fdisk_unref_partitionp) struct fdisk_partition *q = NULL;
+        uint64_t offset, size;
+        int r;
+
+        assert(partno >= 0);
+
+        r = fdisk_get_partition(c, partno, &q);
+        if (r < 0)
+                return log_error_errno(r, "Failed to read created partition metadata: %m");
+
+        assert(fdisk_partition_has_start(q));
+        offset = fdisk_partition_get_start(q);
+        if (offset > UINT64_MAX / 512U)
+                return log_error_errno(SYNTHETIC_ERRNO(ERANGE), "Partition offset too large.");
+
+        assert(fdisk_partition_has_size(q));
+        size = fdisk_partition_get_size(q);
+        if (size > UINT64_MAX / 512U)
+                return log_error_errno(SYNTHETIC_ERRNO(ERANGE), "Partition size too large.");
+
+        *ret_offset = offset * 512U;
+        *ret_size = size * 512U;
+
+        return 0;
+}
```

Comment:
any chance you can split this into a separate preparation commit? i.e. that just splits out these two functions from make_partition_table() without adding the new logic just yet?

Reply:
Done